canonical-web-and-design/maas-ui
# Add tag sort, filter, search and pagination

## Done

- Add a segmented button component.
- Add the action bar to Tags to sort, filter and search.
- Add pagination to the table.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit the tags tab under machines.
- Check that you can filter by automatic or manual tags.
- Check that you can search tags.
- Check that you can change the order of the name and updated columns.
- Check that you can paginate the table (you may have to reduce the `TAGS_PER_PAGE` value to less than the number of tags.

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/703.
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/698.
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/697.
Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/696.

## Screenshots
<img width="1352" alt="Screen Shot 2022-03-04 at 12 07 35 pm" src="https://user-images.githubusercontent.com/361637/156686722-38b5b7ac-6e1b-4720-98e8-fa496ea9cf03.png">
